bootstrap/handlers/auth_middleware.go

```diff
@@ -54,6 +56,27 @@ func VaultAuthenticationHandlerFunc(secretProvider interfaces.SecretProviderExt,
 			w := c.Response()
 			authHeader := r.Header.Get("Authorization")
 			lc.Debugf("Authorizing incoming call to '%s' via JWT (Authorization len=%d)", r.URL.Path, len(authHeader))
+
+			zitiCtx := r.Context().Value("zitiContext")
```

Comment:
We should be doing this only when zerotrust is enabled in configuration.

Reply:
this is older and I think is no longer used in favor of the other if block below this one. i need to build all the pieces and test it out along with the `HandleOpenZiti` function that has no apparent use. I think this is a holdover from the original implementation and I think it needs a bit of refactoring.

As to "We should be doing this only when zerotrust is enabled in configuration.". Do you want another function on `SecretProviderExt` such as `IsZeroTrust` ?

Reply:
I added two more interface functions for your consideration. see what you think.
```

	// ZeroTrustEnabled returns whether zero trust principles are enabled
	ZeroTrustEnabled() bool

	// EnableZeroTrust marks the provider as being zero trust enabled
	EnableZeroTrust()
```